File: LeetCodeTest/242.有效的字母异位词.java
import java.util.HashMap;
import java.util.Map;


/*
 * @lc app=leetcode.cn id=242 lang=java
 *
 * [242] 有效的字母异位词
 */

// @lc code=start
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length()!=t.length()) return false;
        Map<Character,Integer> m1 = new HashMap<Character,Integer>();
        Map<Character,Integer> m2 = new HashMap<Character,Integer>();
        for(int i = 0; i < s.length(); i++){
            m1.merge(s.charAt(i),1,Integer::sum);
        }
        for(int i = 0; i < t.length(); i++){
            m2.merge(t.charAt(i),1,Integer::sum);
        }
        for(int i = 0; i < s.length(); i++){
            if(m2.get(s.charAt(i))==null || (m1.get(s.charAt(i))!=m2.get(s.charAt(i))))
                return false;
        }
        return true;

    }
}
// @lc code=end

File: LeetCodeTest/242.有效的字母异位词.cpp
/*
 * @lc app=leetcode.cn id=242 lang=cpp
 *
 * [242] 有效的字母异位词
 */

// @lc code=start
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length()!=t.length()) return false;
        int c1[26]={0};
        int c2[26]={0};
        for(char c:s){
            c1[c-'a']+=1;
        }
        for(char c:t){
            c2[c-'a']+=1;
        }
        for(char c:s){
            if(c1[c-'a']!=c2[c-'a']) return false;
            
        }

        return true;
    }
};
// @lc code=end

